Index: Yasgui/packages/yasr/src/plugins/table/index.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Yasgui/packages/yasr/src/plugins/table/index.ts b/Yasgui/packages/yasr/src/plugins/table/index.ts
--- a/Yasgui/packages/yasr/src/plugins/table/index.ts	(revision defcb6526807e65732c9e49cd15fdff02fab53f4)
+++ b/Yasgui/packages/yasr/src/plugins/table/index.ts	(revision aa491bed97a739b01f581db8fb025cf0cc9f8b72)
@@ -356,7 +356,8 @@
     ellipseSwitchComponent.addEventListener("change", this.handleSetEllipsisToggle);
     this.tableEllipseSwitch.type = "checkbox";
     ellipseSwitchComponent.appendChild(this.tableEllipseSwitch);
-    this.tableEllipseSwitch.defaultChecked = this.persistentConfig.isEllipsed !== false;
+    // isEllipsed should be unchecked by default
+    this.tableEllipseSwitch.defaultChecked = this.persistentConfig.isEllipsed !== undefined ? this.persistentConfig.isEllipsed : false;
     this.tableControls.appendChild(ellipseToggleWrapper);
 
     // Create table filter
