Subject: [PATCH] GDB-8286: The `ontotext-yasgui-web-component` opens slowly when there are many results.
---
Index: Yasgui/packages/yasr/src/extended-yasr.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Yasgui/packages/yasr/src/extended-yasr.ts b/Yasgui/packages/yasr/src/extended-yasr.ts
--- a/Yasgui/packages/yasr/src/extended-yasr.ts	(revision 591ca74fb98ad5a4edd329b242e81c427b5aefd9)
+++ b/Yasgui/packages/yasr/src/extended-yasr.ts	(revision 9cd0b6407f417287d27e164f2bcf5d9905033b7c)
@@ -22,8 +22,8 @@
     this.persistentJson = persistentJson;
     this.externalPluginsConfigurations = conf.externalPluginsConfigurations;
     if (yasqe.config.paginationOn) {
-      this.yasqe.on("queryResponse", this.draw.bind(this));
-      this.yasqe.on("totalElementsPersisted", this.draw.bind(this));
+      this.yasqe.on("queryResponse", this.updatePaginationRelatedElements.bind(this));
+      this.yasqe.on("totalElementsPersisted", this.updatePaginationRelatedElements.bind(this));
     }
     this.yasqe.on("countAffectedRepositoryStatementsPersisted", this.updateResponseInfo.bind(this));
   }
@@ -67,12 +67,12 @@
   }
 
   draw() {
-    if (this.yasqe.isUpdateQuery() || this.yasqe.isAskQuery() || this.results?.hasError()) {
-      this.hidePluginElementVisibility();
-    } else {
-      this.showPluginElementVisibility();
-    }
-    super.draw();
+    // The rendering of YASR is synchronous and can take time, especially when populating numerous results.
+    // Setting a timeout resolves the visualization of other components without waiting for YASR to finish drawing.
+    setTimeout(() => {
+      this.updatePluginElementVisibility();
+      super.draw();
+    }, 0);
   }
 
   updatePluginSelectorNames() {
@@ -80,6 +80,20 @@
     this.yasrToolbarManagers?.forEach((manager) => manager.updateElement(this));
   }
 
+  private updatePaginationRelatedElements(): void {
+    this.updatePluginElementVisibility();
+    this.updateResponseInfo();
+    this.updatePluginSelectorNames();
+  }
+
+  private updatePluginElementVisibility(): void {
+    if (this.yasqe.isUpdateQuery() || this.yasqe.isAskQuery() || this.results?.hasError()) {
+      this.hidePluginElementVisibility();
+    } else {
+      this.showPluginElementVisibility();
+    }
+  }
+
   private hidePluginElementVisibility() {
     const pluginElement = this.getPluginSelectorsEl();
     if (pluginElement) {
