Index: Yasgui/packages/yasqe/src/defaults.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Yasgui/packages/yasqe/src/defaults.ts b/Yasgui/packages/yasqe/src/defaults.ts
--- a/Yasgui/packages/yasqe/src/defaults.ts	(revision b133c011540f19fea83e04b468a012409c57a436)
+++ b/Yasgui/packages/yasqe/src/defaults.ts	(revision 65a29b30b1121490a64d363cea7f73204027aa4b)
@@ -45,6 +45,10 @@
         const yasqe: Yasqe = _yasqe;
         yasqe.autocomplete();
       },
+      "Alt-Enter": function (_yasqe: any) {
+        const yasqe: Yasqe = _yasqe;
+        yasqe.autocomplete();
+      },
       "Shift-Ctrl-K": function (_yasqe: any) {
         const yasqe: Yasqe = _yasqe;
         const lineNumber = yasqe.getDoc().getCursor().line;
