Index: Yasgui/packages/yasr/src/index.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Yasgui/packages/yasr/src/index.ts b/Yasgui/packages/yasr/src/index.ts
--- a/Yasgui/packages/yasr/src/index.ts	(revision 72c065288c202650825724d2c81776a30c79070c)
+++ b/Yasgui/packages/yasr/src/index.ts	(revision c2573684b43228107fbc50427c9e668977154ace)
@@ -94,6 +94,28 @@
       this.setResponse(resp, undefined, undefined, undefined, undefined, undefined, draw);
     }
   }
+
+  public showWarning(message: string) {
+    this.hideWarning();
+    if (message) {
+        const alertBoxEl = document.createElement('alert-box');
+        // @ts-ignore
+        alertBoxEl.message = message;
+        // @ts-ignore
+        alertBoxEl.isVisible = true;
+        // @ts-ignore
+        alertBoxEl.type = 'warning';
+        this.rootEl.prepend(alertBoxEl);
+    }
+  }
+
+  public hideWarning() {
+      let alertBoxEl = this.rootEl.querySelector('alert-box');
+      if (alertBoxEl) {
+          alertBoxEl.remove();
+      }
+  }
+
   private getConfigFromStorage() {
     const storageId = this.getStorageId(this.config.persistenceLabelConfig);
     if (storageId) {
@@ -275,6 +297,7 @@
     return {};
   }
   public selectPlugin(plugin: string) {
+    this.hideWarning();
     if (this.selectedPlugin === plugin) {
       // Don't re-render when selecting the same plugin. Also see #1893
       return;
