Subject: [PATCH] GDB-8188: Buttons labels of dialogs are not translated
---
Index: Yasgui/packages/yasgui/src/Tab.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Yasgui/packages/yasgui/src/Tab.ts b/Yasgui/packages/yasgui/src/Tab.ts
--- a/Yasgui/packages/yasgui/src/Tab.ts	(revision b453bf272eee176999d0d7d74215d3e66c5459de)
+++ b/Yasgui/packages/yasgui/src/Tab.ts	(revision 15858c02bcb2c07b66dd6acbfeab1b4e5af68f4e)
@@ -674,7 +674,7 @@
       },
       requestConfig: yasgui ? yasgui.config.requestConfig : { ...Yasgui.defaults.requestConfig },
       id: getRandomId(),
-      name: yasgui ? yasgui.createTabName() : Yasgui.defaults.tabName,
+      name: yasgui ? yasgui.createTabName() : "",
     };
   }
 }
Index: Yasgui/packages/yasgui/src/defaults.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Yasgui/packages/yasgui/src/defaults.ts b/Yasgui/packages/yasgui/src/defaults.ts
--- a/Yasgui/packages/yasgui/src/defaults.ts	(revision b453bf272eee176999d0d7d74215d3e66c5459de)
+++ b/Yasgui/packages/yasgui/src/defaults.ts	(revision 15858c02bcb2c07b66dd6acbfeab1b4e5af68f4e)
@@ -32,7 +32,7 @@
       }
       return "yagui_" + id;
     },
-    tabName: "Query",
+    defaultTabNameLabelKey: "yasgui.tab_list.tab.default.name",
     corsProxy: undefined,
     persistencyExpire: 60 * 60 * 24 * 30,
     persistenceLabelResponse: "response",
Index: Yasgui/packages/yasgui/src/index.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Yasgui/packages/yasgui/src/index.ts b/Yasgui/packages/yasgui/src/index.ts
--- a/Yasgui/packages/yasgui/src/index.ts	(revision b453bf272eee176999d0d7d74215d3e66c5459de)
+++ b/Yasgui/packages/yasgui/src/index.ts	(revision 15858c02bcb2c07b66dd6acbfeab1b4e5af68f4e)
@@ -37,7 +37,7 @@
   autofocus: boolean;
   endpointInfo: ((tab?: Tab) => Element) | undefined;
   copyEndpointOnNewTab: boolean;
-  tabName: string;
+  defaultTabNameLabelKey: string;
   corsProxy: string | undefined;
   endpointCatalogueOptions: EndpointSelectConfig<EndpointObject>;
   //The function allows us to modify the config before we pass it on to a tab
@@ -207,7 +207,7 @@
     return persistenceId(this) + "_" + label;
   }
   public createTabName(name?: string, i: number = 0) {
-    if (!name) name = this.config.tabName;
+    if (!name) name = this.translationService.translate(this.config.defaultTabNameLabelKey);
     var fullName = name + (i > 0 ? " " + i : "");
     if (this.tabNameTaken(fullName)) fullName = this.createTabName(name, i + 1);
     return fullName;
