Subject: [PATCH] Default message info not displayed when the reinitialized function is called.
---
Index: Yasgui/packages/utils/src/yasgui-configuration.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Yasgui/packages/utils/src/yasgui-configuration.ts b/Yasgui/packages/utils/src/yasgui-configuration.ts
--- a/Yasgui/packages/utils/src/yasgui-configuration.ts	(revision 336cfd714d7a8dabb93ced91996102fc5767a946)
+++ b/Yasgui/packages/utils/src/yasgui-configuration.ts	(revision 788df97bea87f006f0eec94a585b623350dae229)
@@ -2,5 +2,6 @@
   yasguiConfig?: {
     infer: boolean;
     sameAs: boolean;
+    clearState: boolean;
   }
 }
Index: Yasgui/packages/yasgui/src/Tab.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Yasgui/packages/yasgui/src/Tab.ts b/Yasgui/packages/yasgui/src/Tab.ts
--- a/Yasgui/packages/yasgui/src/Tab.ts	(revision 336cfd714d7a8dabb93ced91996102fc5767a946)
+++ b/Yasgui/packages/yasgui/src/Tab.ts	(revision 788df97bea87f006f0eec94a585b623350dae229)
@@ -680,6 +680,7 @@
     yasrConf.downloadAsOptions = this.yasgui.config.yasr.downloadAsOptions;
     yasrConf.showResultInfo = this.yasgui.config.yasr.showResultInfo;
     yasrConf.showQueryLoader = this.yasgui.config.yasr.showQueryLoader;
+    yasrConf.clearState = this.yasgui.config.clearState;
     yasrConf.tabId = this.getId();
 
     if (this.yasqe) {
Index: Yasgui/packages/yasgui/src/defaults.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Yasgui/packages/yasgui/src/defaults.ts b/Yasgui/packages/yasgui/src/defaults.ts
--- a/Yasgui/packages/yasgui/src/defaults.ts	(revision 336cfd714d7a8dabb93ced91996102fc5767a946)
+++ b/Yasgui/packages/yasgui/src/defaults.ts	(revision 788df97bea87f006f0eec94a585b623350dae229)
@@ -71,5 +71,6 @@
     autoAddOnInit: true,
     requestConfig: Yasqe.defaults.requestConfig,
     contextMenuContainer: undefined,
+    clearState: false,
   };
 }
Index: Yasgui/packages/yasgui/src/index.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Yasgui/packages/yasgui/src/index.ts b/Yasgui/packages/yasgui/src/index.ts
--- a/Yasgui/packages/yasgui/src/index.ts	(revision 336cfd714d7a8dabb93ced91996102fc5767a946)
+++ b/Yasgui/packages/yasgui/src/index.ts	(revision 788df97bea87f006f0eec94a585b623350dae229)
@@ -42,6 +42,7 @@
   endpointInfo: ((tab?: Tab) => Element) | undefined;
   copyEndpointOnNewTab: boolean;
   defaultTabNameLabelKey: string;
+  clearState: boolean;
   corsProxy: string | undefined;
   endpointCatalogueOptions: EndpointSelectConfig<EndpointObject>;
   //The function allows us to modify the config before we pass it on to a tab
Index: Yasgui/packages/yasr/src/defaults.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Yasgui/packages/yasr/src/defaults.ts b/Yasgui/packages/yasr/src/defaults.ts
--- a/Yasgui/packages/yasr/src/defaults.ts	(revision 336cfd714d7a8dabb93ced91996102fc5767a946)
+++ b/Yasgui/packages/yasr/src/defaults.ts	(revision 788df97bea87f006f0eec94a585b623350dae229)
@@ -29,5 +29,6 @@
     plugins: {},
     pluginOrder: ["table", "response"], // Default plugins, others are sorted alphabetically
     defaultPlugin: "table",
+    clearState: false
   };
 }
Index: Yasgui/packages/yasr/src/extended-yasr.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Yasgui/packages/yasr/src/extended-yasr.ts b/Yasgui/packages/yasr/src/extended-yasr.ts
--- a/Yasgui/packages/yasr/src/extended-yasr.ts	(revision 336cfd714d7a8dabb93ced91996102fc5767a946)
+++ b/Yasgui/packages/yasr/src/extended-yasr.ts	(revision 788df97bea87f006f0eec94a585b623350dae229)
@@ -120,7 +120,7 @@
       const bindings = this.results.getBindings() || [];
       return bindings.length > 0;
     }
-    return true;
+    return false;
   }
 
   getTabId(): string | undefined {
Index: Yasgui/packages/yasr/src/index.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Yasgui/packages/yasr/src/index.ts b/Yasgui/packages/yasr/src/index.ts
--- a/Yasgui/packages/yasr/src/index.ts	(revision 336cfd714d7a8dabb93ced91996102fc5767a946)
+++ b/Yasgui/packages/yasr/src/index.ts	(revision 788df97bea87f006f0eec94a585b623350dae229)
@@ -85,12 +85,17 @@
     this.rootEl.appendChild(this.resultsEl);
     this.initializePlugins();
 
-    const resp = data || this.getResponseFromStorage();
+    let resp;
+    if (!this.config.clearState) {
+        resp = data || this.getResponseFromStorage();
+    }
     if (resp) {
       // Set response without draw yasr to prevent double rendering yasr active plugin. The drawing function of the plugin that displays the response
       // will be called when it is marked as active.
       const draw = false;
       this.setResponse(resp, undefined, undefined, undefined, undefined, undefined, draw);
+    } else {
+        this.draw();
     }
     this.drawHeader();
   }
@@ -818,6 +823,7 @@
    * overwrite or explicitly call previously added or default ones.
    */
   errorRenderers?: ((error: Parser.ErrorSummary) => Promise<HTMLElement | undefined>)[];
+  clearState: boolean;
 }
 
 export function registerPlugin(name: string, plugin: typeof Plugin, enable = true) {
