Subject: [PATCH] GDB-9318: Scrollbar in Sparql editor (Yasqe)
---
Index: Yasgui/packages/yasqe/src/index.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Yasgui/packages/yasqe/src/index.ts b/Yasgui/packages/yasqe/src/index.ts
--- a/Yasgui/packages/yasqe/src/index.ts	(revision e2011d8f63e3eb50bdd04fba521af196ea0fce8f)
+++ b/Yasgui/packages/yasqe/src/index.ts	(revision 36dcc3617c659039a65fa8ca126246e87df61358)
@@ -524,7 +524,8 @@
     addClass(chip, "resizeChip");
     this.resizeWrapper.appendChild(chip);
     this.resizeWrapper.addEventListener("mousedown", this.initDrag, false);
-    this.resizeWrapper.addEventListener("dblclick", this.expandEditor);
+    // We don't want this behavior, when the component is used with other components in a view, this behavior can make the view look strange.
+    // this.resizeWrapper.addEventListener("dblclick", this.expandEditor);
     this.rootEl.appendChild(this.resizeWrapper);
   }
   private drawKeyboardShortcutsButton() {
@@ -1242,7 +1243,7 @@
     this.abortQuery();
     this.unregisterEventListeners();
     this.resizeWrapper?.removeEventListener("mousedown", this.initDrag, false);
-    this.resizeWrapper?.removeEventListener("dblclick", this.expandEditor);
+    // this.resizeWrapper?.removeEventListener("dblclick", this.expandEditor);
     for (const autocompleter in this.autocompleters) {
       this.disableCompleter(autocompleter);
     }
