Subject: [PATCH] Skip aborting the query if the "onQueryAborted" function is passed to the configuration
---
Index: Yasgui/packages/yasqe/src/index.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Yasgui/packages/yasqe/src/index.ts b/Yasgui/packages/yasqe/src/index.ts
--- a/Yasgui/packages/yasqe/src/index.ts	(revision 55b586a4262ce217b9bb37bf56360e021ddead6e)
+++ b/Yasgui/packages/yasqe/src/index.ts	(revision 1e9f064aa484ea6b9873561af73580063ab68fa8)
@@ -562,9 +562,12 @@
       }
       this.isQueryAborted = true;
       const req = this.req;
-      this.abortQuery();
+
       if (this.config.onQueryAborted instanceof Function) {
+        this.queryStateChanged(true, true);
         this.config.onQueryAborted(req).finally(() => this.queryStateChanged(false, false));
+      } else {
+        this.abortQuery();
       }
     };
 
