Index: Yasgui/packages/yasgui/src/Tab.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Yasgui/packages/yasgui/src/Tab.ts b/Yasgui/packages/yasgui/src/Tab.ts
--- a/Yasgui/packages/yasgui/src/Tab.ts	(revision 289fc5018b5bb7fa6d9682c075a8769b84ba7f1d)
+++ b/Yasgui/packages/yasgui/src/Tab.ts	(revision 4e19d75dceabc791a493da5b20d4387e48dee296)
@@ -689,6 +689,7 @@
     yasrConf.tabId = this.getId();
     yasrConf.fullscreen = this.yasgui.config.yasr.fullscreen;
     yasrConf.showFullscreenButton = this.yasgui.config.yasr.showFullscreenButton;
+    yasrConf.themeName = this.yasgui.config.yasr.themeName;
     if (this.yasgui.config.yasr.selectedPlugin != null) {
         yasrConf.selectedPlugin = this.yasgui.config.yasr.selectedPlugin;
     }
Index: Yasgui/packages/yasr/src/plugins/response/index.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Yasgui/packages/yasr/src/plugins/response/index.ts b/Yasgui/packages/yasr/src/plugins/response/index.ts
--- a/Yasgui/packages/yasr/src/plugins/response/index.ts	(revision 289fc5018b5bb7fa6d9682c075a8769b84ba7f1d)
+++ b/Yasgui/packages/yasr/src/plugins/response/index.ts	(revision 4e19d75dceabc791a493da5b20d4387e48dee296)
@@ -96,6 +96,9 @@
     }
 
     this.cm = CodeMirror(this.yasr.resultsEl, codemirrorOpts);
+    if (this.cm) {
+      this.cm.setOption("theme", this.yasr.config.themeName);
+    }
     // Don't show less originally we've already set the value in the codemirrorOpts
     if (lines.length > config.maxLines) this.showLess(false);
   }
Index: Yasgui/packages/yasr/src/index.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Yasgui/packages/yasr/src/index.ts b/Yasgui/packages/yasr/src/index.ts
--- a/Yasgui/packages/yasr/src/index.ts	(revision 289fc5018b5bb7fa6d9682c075a8769b84ba7f1d)
+++ b/Yasgui/packages/yasr/src/index.ts	(revision 4e19d75dceabc791a493da5b20d4387e48dee296)
@@ -834,6 +834,7 @@
   sparqlResponse?: string;
   fullscreen?: boolean;
   showFullscreenButton?: boolean;
+  themeName?: string;
   selectedPlugin?: string;
   /**
    * Custom renderers for errors.
