Subject: [PATCH] GDB-14270: Expose full-screen mode
---
Index: Yasgui/packages/yasgui/src/Tab.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Yasgui/packages/yasgui/src/Tab.ts b/Yasgui/packages/yasgui/src/Tab.ts
--- a/Yasgui/packages/yasgui/src/Tab.ts	(revision 2bee846a74cb04a75d1ff8990f0eb91ca8e3c33a)
+++ b/Yasgui/packages/yasgui/src/Tab.ts	(revision 5be3b444894fc648fb39a2c4dae718c7a0f4fe2e)
@@ -683,6 +683,7 @@
     yasrConf.clearState = this.yasgui.config.clearState;
     yasrConf.isExplainPlan = this.yasgui.config.yasr.isExplainPlan
     yasrConf.tabId = this.getId();
+    yasrConf.yasrFullscreen = this.yasgui.config.yasr.yasrFullscreen;
 
     if (this.yasqe) {
       if (this.yasgui.config.yasr.sparqlResponse) {
Index: Yasgui/packages/yasr/src/extended-yasr.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Yasgui/packages/yasr/src/extended-yasr.ts b/Yasgui/packages/yasr/src/extended-yasr.ts
--- a/Yasgui/packages/yasr/src/extended-yasr.ts	(revision 2bee846a74cb04a75d1ff8990f0eb91ca8e3c33a)
+++ b/Yasgui/packages/yasr/src/extended-yasr.ts	(revision 5be3b444894fc648fb39a2c4dae718c7a0f4fe2e)
@@ -10,6 +10,11 @@
 
   private yasrToolbarManagers: YasrToolbarPluginManager[] | undefined;
   private readonly persistentJson: any;
+  private isFullscreen = false;
+  private fullscreenButton: HTMLButtonElement | undefined;
+  private fullscreenIcon: HTMLElement | undefined;
+  private boundFullscreenChangeHandler: EventListener | undefined;
+  private boundEscapeHandler: EventListener | undefined;
 
   constructor(yasqe: Yasqe, parent: HTMLElement, conf: Partial<Config> = {}, persistentJson?: any) {
     super(yasqe, parent, conf, persistentJson?.yasr.response);
@@ -20,6 +25,11 @@
       this.yasqe.on("totalElementsPersisted", this.updatePaginationRelatedElements.bind(this));
     }
     this.yasqe.on("countAffectedRepositoryStatementsPersisted", this.updateResponseInfo.bind(this));
+
+    // Toggle only if the default fullscreen value is true and YASR is not in fullscreen mode.
+    if (this.config.yasrFullscreen?.defaultFullscreen && !this.isFullscreen) {
+        this.toggleFullscreen();
+    }
   }
 
   //=================================
@@ -41,6 +51,14 @@
       return;
     }
     const pluginSelectorsEl = this.getPluginSelectorsEl();
+    this.fullscreenButton = this.createFullscreenButton();
+    pluginSelectorsEl.appendChild(this.fullscreenButton);
+
+    if (this.config.yasrFullscreen?.allowEscape) {
+        this.boundEscapeHandler = this.onEscapeHandler.bind(this);
+        document.addEventListener('keydown', this.boundEscapeHandler);
+    }
+
     const spacerElement = document.createElement("li");
     spacerElement.classList.add("spacer");
     pluginSelectorsEl.appendChild(spacerElement);
@@ -127,6 +145,13 @@
   public destroy() {
     super.destroy();
     this.yasrToolbarManagers?.forEach((manager) => manager.destroy(this));
+    if (this.boundFullscreenChangeHandler) {
+        this.fullscreenButton?.removeEventListener('click', this.boundFullscreenChangeHandler);
+    }
+
+    if (this.boundEscapeHandler) {
+        document.removeEventListener('keydown', this.boundEscapeHandler);
+    }
   }
 
   private getUpdateTypeQueryResponseInfo(): string {
@@ -282,6 +307,60 @@
     }
     return "";
   }
+
+  /**
+   * Handles global keyboard events and exits fullscreen when the Escape key is pressed.
+   *
+   * @param event - The DOM event triggered by a key interaction.
+   */
+  private onEscapeHandler(event: Event): void {
+      if (event instanceof KeyboardEvent && event.key === 'Escape' && this.isFullscreen) {
+          this.toggleFullscreen();
+      }
+  }
+
+  /**
+   * Creates the icon element used inside the fullscreen toggle button.
+   *
+   * @returns The <i> element representing the fullscreen icon.
+   */
+  private createFullscreenIcon(): HTMLElement {
+      const fullscreenIcon = document.createElement('i');
+      fullscreenIcon.classList.add('ri-fullscreen-line', 'ri-1x');
+      return fullscreenIcon;
+  }
+
+  /**
+   * Creates the fullscreen toggle button and attaches the click event listener.
+   *
+   * @returns The button element used to toggle fullscreen mode.
+   */
+  private createFullscreenButton(): HTMLButtonElement {
+      const fullscreenButton = document.createElement('button');
+      fullscreenButton.classList.add('yasr-fullscreen-button');
+
+      this.boundFullscreenChangeHandler = this.toggleFullscreen.bind(this);
+      fullscreenButton.addEventListener('click', this.boundFullscreenChangeHandler);
+
+      this.fullscreenIcon = this.createFullscreenIcon();
+      fullscreenButton.appendChild(this.fullscreenIcon);
+
+      return fullscreenButton;
+  }
+
+  /**
+   * Toggles the fullscreen mode and emits a "fullscreen-changed" event with the new mode.
+   */
+  private toggleFullscreen(): void {
+      this.isFullscreen = !this.isFullscreen;
+
+      this.rootEl.classList.toggle('yasr-fullscreen', this.isFullscreen);
+
+      this.fullscreenIcon?.classList.toggle('ri-fullscreen-line', !this.isFullscreen);
+      this.fullscreenIcon?.classList.toggle('ri-fullscreen-exit-line', this.isFullscreen);
+
+      this.emit('fullscreen-changed', this.isFullscreen);
+    }
 }
 
 export interface YasrToolbarPlugin {
Index: Yasgui/packages/yasr/src/index.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Yasgui/packages/yasr/src/index.ts b/Yasgui/packages/yasr/src/index.ts
--- a/Yasgui/packages/yasr/src/index.ts	(revision 2bee846a74cb04a75d1ff8990f0eb91ca8e3c33a)
+++ b/Yasgui/packages/yasr/src/index.ts	(revision 5be3b444894fc648fb39a2c4dae718c7a0f4fe2e)
@@ -32,6 +32,7 @@
   emit(event: "drawn", instance: Yasr, plugin: Plugin<any>): boolean;
   on(event: "toggle-help", listener: (instance: Yasr) => void): this;
   emit(event: "toggle-help", instance: Yasr): boolean;
+  emit(event: 'fullscreen-changed', isFullscreen: boolean): boolean
 }
 export class Yasr extends EventEmitter {
   public results?: Parser;
@@ -818,6 +819,10 @@
   showResultInfo: boolean;
   showQueryLoader: boolean;
   sparqlResponse?: string;
+  yasrFullscreen?: {
+      defaultFullscreen: boolean,
+      allowEscape: boolean,
+  };
   /**
    * Custom renderers for errors.
    * Allow multiple to be able to add new custom renderers without having to
