Subject: [PATCH] GDB-13833: Support fullscreen mode for the yasgui sparql editor
---
Index: Yasgui/packages/yasqe/src/index.ts
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Yasgui/packages/yasqe/src/index.ts b/Yasgui/packages/yasqe/src/index.ts
--- a/Yasgui/packages/yasqe/src/index.ts	(revision 69fb9acefdc01ac4feeb94c5586ec5126194117b)
+++ b/Yasgui/packages/yasqe/src/index.ts	(revision 541cace35347f6ccb9deeb88b84c72ac841b8c2c)
@@ -516,8 +516,8 @@
 
       this.queryBtn.onclick = () => {
         if (this.config.queryingDisabled) return; // Don't do anything
-          this.pageNumber = 1;
-          this.query().catch(() => {}); //catch this to avoid unhandled rejection
+        this.pageNumber = 1;
+        this.query().catch(() => {}); //catch this to avoid unhandled rejection
       };
 
       const querySplitButtonEl = document.createElement("query-split-button");
@@ -1287,6 +1287,17 @@
     this.setSize(null, "100%");
   }
 
+  /**
+   * Registers a callback to be executed on destroy.
+   *
+   * @param destroyCallback Callback invoked when the instance is destroyed.
+   */
+  public addDestroyCallback(destroyCallback: () => void) {
+    if (typeof destroyCallback === "function") {
+      this.subscriptions.push(destroyCallback);
+    }
+  }
+
   public destroy() {
     this.subscriptions.forEach((subscription) => subscription());
     //  Abort running query;
